platform/darwin/docs/guides/Style Layers Using Expressions.md.ejs

```diff
@@ -0,0 +1,168 @@
+<%- guideExample(guide, 'Exponential', os) %>
+
+### Interval
+
+Steps, or intervals, create a range using the keys from the stops dictionary. The range is from the given key to just less than the next key. The attribute values that fall into that range are then styled using the layout or paint value assigned to that key.
+
+When we use the stops dictionary given above with an interval interpolation mode, we create ranges where earthquakes with a magnitude of 0 to just less than 2.5 would be yellow, 2.5 to just less than 5 would be orange, and so on.
+<%- guideExample(guide, 'Interval', os) %>
+
+![interval mode](img/data-driven-styling/interval.png)
+
+### Categorical
+
+At each stop, a categorical, or match expression, produces an output value equal to the function input. We’re going to match different feature attribute and style layer property values than we did for the previous two modes.
+
+There are three main types of events in the USGS dataset: earthquakes, explosions, and quarry blasts. In this case, the color of the circle layer will be determined by the type of event, with a default value of blue to catch any events that do not fall into any of those categories.
+
+<%- guideExample(guide, 'Categorical', os) %>
+
+`MGL_MATCH` takes the attribute key, followed by possible attribute and style value pairs. The final argument is a default style value.
+
+![categorical mode](img/data-driven-styling/categorical1.png) ![categorical mode](img/data-driven-styling/categorical2.png)
+
+### Identity
+
+Identity interpolation mode uses the attribute’s value as the style layer property value. For example, you can set the `circleRadius` to the earthquake’s magnitude. Since the attribute value itself will be used as the style value, `sourceStops` should be set to `nil`.
+
+<%- guideExample(guide, 'Identity', os) %>
+
+![identity mode](img/data-driven-styling/identity.png)
+
+Some built-in expressions functions can be applied to attribute values to style layer property values. To set the circle radius to 3 times the earthquake’s magnitude, create a `multiply:by` function that takes the attribute value and the multiplier as arguments.
+
+<%- guideExample(guide, 'Multiply', os) %>
+
+![multiply magnitude](img/data-driven-styling/multiply.png)
+
+Through expressions, you can also cast attribute values in order to use them. One example is to cast an integer as an `NSString` and use it as a text value.
+
+<%- guideExample(guide, 'Cast', os) %>
+
+![cast a value](img/data-driven-styling/cast.png)
+
+<% if (iOS) { -%>
+
+  ##Migrating from Style Functions to Expressions
+
+  For constant values that do not necessarily change based on camera or attribute values, use `[NSExpression expressionForConstantValue:]` (previously `[MGLStyleValue valueWithRawValue:]`). Expressions support format string syntax, as shown above, as well as convenience methods that include those listed below.
```

Comment:
We should probably note that the “Current syntax” column recommends convenience initializers to ease the transition from MGLStyleFunction, whereas other guides and examples rely more heavily on expression format strings.